shards/shards.go

```diff
@@ -667,13 +667,71 @@ search:
 			r.Priority = r.priority
 			r.MaxPendingPriority = pending.max()
 
-			sender.Send(r.SearchResult)
+			// For simple shards, send the results immediately.
+			if len(r.SearchResult.RepoURLs) <= 1 {
+				sender.Send(r.SearchResult)
+				break
+			}
+
+			// For compound shards, stream results per repo.
+			stats, srs := splitByRepository(r.SearchResult)
+			tr.LazyPrintf("compound shard: results from %d repositories", len(srs))
+			for _, sr := range srs {
+				if len(sr.Files) == 0 {
+					panic("unreachable")
+				}
+				sr.Priority = sr.Files[0].RepositoryPriority
+
+				// TODO (stefan): make sure repos in compound shards are searched in order of their priority.
+				sr.MaxPendingPriority = r.MaxPendingPriority
+				sender.Send(sr)
+			}
+			sender.Send(&zoekt.SearchResult{Stats: stats})
```

Comment:
bug: you are not setting Priority or MaxPendingPriority here

Reply:
why? We already send events like this, see EG stream/stream.go:108. I think we should assume the caller can handle it, but maybe I am missing something?

Reply:
Sorry, I wasn't aware the caller handled these fields not being set. Hope there isn't a lurking bug in horiztonal searcher then :P